File: src/dataset/cpp/allergies_exercism-Richard-m-j/claude/ClaudeSonnet4_allergies_v3.cpp
#include "allergies.h"

namespace allergies
{
    static constexpr int ALLERGEN_COUNT = 8;
    static constexpr const char* ALLERGEN_NAMES[ALLERGEN_COUNT] = {
        "eggs", "peanuts", "shellfish", "strawberries",
        "tomatoes", "chocolate", "pollen", "cats"
    };

    bool allergy_test::is_allergic_to(std::string allergen)
    {
        for (int i = 0; i < ALLERGEN_COUNT; ++i) {
            if (allergen == ALLERGEN_NAMES[i]) {
                return (allergy_score & (1 << i)) != 0;
            }
        }
        return false;
    }

    std::unordered_set<std::string> allergy_test::get_allergies()
    {
        std::unordered_set<std::string> allergies;
        allergies.reserve(ALLERGEN_COUNT);
        
        for (int i = 0; i < ALLERGEN_COUNT; ++i) {
            if (allergy_score & (1 << i)) {
                allergies.insert(ALLERGEN_NAMES[i]);
            }
        }
        return allergies;
    }

    allergy_test::allergy_test(int score)
    {
        allergy_score = score;
    }
}

File: src/dataset/javascript/allergies_exercism-SLIB53/claude/ClaudeSonnet4_allergies_v3.js
export class Allergies {
  static ALLERGENS = [
    'eggs',
    'peanuts',
    'shellfish',
    'strawberries',
    'tomatoes',
    'chocolate',
    'pollen',
    'cats'
  ];

  static ALLERGEN_MAP = new Map([
    ['eggs', 1],
    ['peanuts', 2],
    ['shellfish', 4],
    ['strawberries', 8],
    ['tomatoes', 16],
    ['chocolate', 32],
    ['pollen', 64],
    ['cats', 128]
  ]);

  constructor(allergyScore) {
    this.allergyScore = allergyScore;
    this._cachedList = null;
  }

  list() {
    if (this._cachedList === null) {
      this._cachedList = [];
      for (let i = 0; i < Allergies.ALLERGENS.length; i++) {
        if (this.allergyScore & (1 << i)) {
          this._cachedList.push(Allergies.ALLERGENS[i]);
        }
      }
    }
    return this._cachedList;
  }

  allergicTo(item) {
    const allergenValue = Allergies.ALLERGEN_MAP.get(item);
    return allergenValue ? (this.allergyScore & allergenValue) !== 0 : false;
  }
}